src/main/java/edu/ksu/canvas/attendance/submitter/AssignmentSubmitter.java

```diff
@@ -171,23 +192,48 @@ private String getCommentHeader(AttendanceSummaryModel.Entry entry, AttendanceCo
     /**
      * Calculates the grade of one student.
      */
-    private double calculateStudentGrade(AttendanceAssignment attendanceAssignment, AttendanceSummaryModel.Entry entry, boolean isSimpleAttendance) {
-        String presentPoints = attendanceAssignment.getPresentPoints();
-        String tardyPoints = attendanceAssignment.getTardyPoints();
-        String excusedPoints = attendanceAssignment.getExcusedPoints();
-        String absentPoints = attendanceAssignment.getAbsentPoints();
+    private double calculateStudentGrade(AttendanceAssignment attendanceAssignment, List<AttendanceSummaryModel> summaryForSections, boolean isSimpleAttendance, AttendanceStudent student) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "calculateStudentGrade" is 11 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)